tests/test_solver_utils.py

```diff
@@ -113,3 +116,68 @@ def test_vartype_dimod_smoke(self):
 
         with self.assertRaises(ValueError):
             StructuredSolver.reformat_parameters("INTEGER", {}, {})
+
+class QpuAccessTimeEstimate(unittest.TestCase):
+    """Test QPU access time estimation method."""
+
+    def test_qpu_access_time_estimate_mock(self):
+        solver_mock = StructuredSolver(data=mocks.qpu_pegasus_solver_data(16,
+            problem_timing_data=mocks.qpu_problem_timing_data(qpu='advantage')), client=None)
+
+        # not allowed annealing_time together with anneal_schedule
+        with self.assertRaises(ValueError):
+            solver_mock.estimate_qpu_access_time(num_qubits=1000,
+                                            anneal_schedule=[[0.0, 0.0], [50.0, 0.5]],
+                                            annealing_time=150)
+        # reverse anneal without required initial state
+        with self.assertRaises(ValueError):
+            solver_mock.estimate_qpu_access_time(num_qubits=1000,
+                                            anneal_schedule=[[0.0, 1.0], [2.75, 0.45], [82.75, 0.45], [82.761, 1.0]])
+
+        # currently support is for version 1.0.x
+        with self.assertRaises(ValueError):
+            solver_mock_ver = copy.deepcopy(solver_mock)
+            solver_mock_ver.properties["problem_timing_data"]["version"] = '1.1.0'
+            solver_mock_ver.estimate_qpu_access_time(num_qubits=1000)
+
+        with self.assertRaises(KeyError):
+            solver_mock_ver = copy.deepcopy(solver_mock)
+            solver_mock_ver.properties["problem_timing_data"].pop("version")
+            solver_mock_ver.estimate_qpu_access_time(num_qubits=1000)
+
+        for d, t in [({'num_qubits': 1000}, 15341),
+                     ({'num_qubits': 1000, 'num_reads': 500}, 149225),
+                     ({'num_qubits': 1000, 'annealing_time': 400}, 15721),
+                     ({'num_qubits': 1000, 'anneal_schedule': [[0.0, 0.0], [800.0, 0.5]]}, 16121),
+                     ({'num_qubits': 1000, 'anneal_schedule': [[0.0, 1.0], [2, 0.45], [102, 0.45], [102, 1.0]], 'initial_state': {node: 0 for node in solver_mock.nodes}}, 15427),
+                     ]:
+            self.assertEqual(int(solver_mock.estimate_qpu_access_time(**d)), t)
```

Comment:
Good fit for `parameterized` (e.g. see [here](https://github.com/dwavesystems/dwave-cloud-client/blob/442211d5279e9e6a96dc42175549caef66844970/tests/test_mock_submission.py#L1004)).